src/mlpack/methods/linear_svm/linear_svm_main.cpp

```diff
@@ -0,0 +1,417 @@
+    "matrix of data points, if " + PRINT_PARAM_STRING("test") + " is "
+    "specified.  The " + PRINT_PARAM_STRING("test") + " parameter can be "
+    "specified without the " + PRINT_PARAM_STRING("training") + " parameter, "
+    "so long as an existing LinearSVM model is given with the " +
+    PRINT_PARAM_STRING("input_model") + " parameter.  The output predictions "
+    "from the LinearSVM model may be saved with the " +
+    PRINT_PARAM_STRING("predictions") + " parameter." +
+    "\n\n"
+    "As an example, to train a LinaerSVM model on the data '" +
+    PRINT_DATASET("data") + "' with labels '" + PRINT_DATASET("labels") + "' "
+    "with L2 regularization of 0.1, saving the model to '" +
+    PRINT_MODEL("lsvm_model") + "', the following command may be used:"
+    "\n\n" +
+    PRINT_CALL("linear_svm", "training", "data", "labels", "labels",
+        "lambda", 0.1, "delta", 1.0, "number_of_classes", 0,
+        "output_model", "lsvm_model") +
+    "\n\n"
+    "Then, to use that model to predict classes for the dataset '" +
+    PRINT_DATASET("test") + "', storing the output predictions in '" +
+    PRINT_DATASET("predictions") + "', the following command may be used: "
+    "\n\n" +
+    PRINT_CALL("linear_svm", "input_model", "lsvm_model", "test", "test",
+        "predictions", "predictions"),
+    SEE_ALSO("@random_forest", "#random_forest"),
+    SEE_ALSO("@logistic_regression", "#logistic_regression"),
+    SEE_ALSO("LinearSVM on Wikipedia",
+        "https://en.wikipedia.org/wiki/Support-vector_machine"),
+    SEE_ALSO("mlpack::svm::LinearSVM C++ class documentation",
+        "@doxygen/classmlpack_1_1svm_1_1LinearSVM.html"));
+
+// Training parameters.
+PARAM_MATRIX_IN("training", "A matrix containing the training set (the matrix "
+    "of predictors, X).", "t");
+PARAM_UROW_IN("labels", "A matrix containing labels (0 or 1) for the points "
+    "in the training set (y).", "l");
+
+// Optimizer parameters.
+PARAM_DOUBLE_IN("lambda", "L2-regularization parameter for training.", "L",
+    0.0001);
+PARAM_DOUBLE_IN("delta", "Margin of difference between correct class and other "
+    "classes.", "d", 1.0);
+PARAM_INT_IN("number_of_classes", "Number of classes for classification; if "
+    "unspecified (or 0), the number of classes found in the labels will be "
+    "used.", "c", 0);
+PARAM_FLAG("no_intercept", "Do not add the intercept term to the model.", "N");
+PARAM_STRING_IN("optimizer", "Optimizer to use for training ('lbfgs' or "
+    "'psgd').", "O", "lbfgs");
+PARAM_DOUBLE_IN("tolerance", "Convergence tolerance for optimizer.", "e",
+    1e-10);
+PARAM_INT_IN("max_iterations", "Maximum iterations for optimizer (0 indicates "
+    "no limit).", "n", 10000);
+#ifdef HAS_OPENMP
+PARAM_DOUBLE_IN("step_size", "Step size for ParallelSGD optimizer.", "s", 0.01);
+PARAM_FLAG("shuffle", "Don't shuffle the order in which data points are "
+    "visited for ParallelSGD.", "S");
+#endif
```

Comment:
I think it should be okay to enable these even if `HAS_OPENMP` is not set.  If we use preprocessor macros to enable or disable some options, it actually makes the documentation really hard for the website, since we would need some way to clearly specify which conditions are needed for a certain option to exist (which we currently don't have, and preferably I'd like to avoid creating :)).

Reply:
Done.